src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +297,171 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
+	}
+
+
+	private void verifyResponseForValidRequests(int code, String response, String rawResponse) {
+		if(Config.debug)
+		{
+		    System.out.println("DEBUG Worker[" + threadId + "]: Base Case Check " + url.toString());
+		}
+
+
+		//TODO move this option to the Adv options
+		//if the response does not match the base case
+		Pattern regexFindFile = Pattern.compile(".*file not found.*", Pattern.CASE_INSENSITIVE);
+
+		Matcher m = regexFindFile.matcher(response);
+
+		//need to clean the base case of the item we are looking for
+		String basecase = FilterResponce.removeItemCheckedFor(work.getBaseCaseObj().getBaseCase(), work.getItemToCheck());
+
+		if(m.find())
+		{
+			System.out.println("DEBUG Worker[" + threadId + "]: 404 for: " + url.toString());
+		}
+		else if(!response.equalsIgnoreCase(basecase))
+		{
+			notifyItemFound(code, response, rawResponse, basecase);
+		}
+	}
+	
+	private void notifyItemFound(int code, String response, String rawResponse, String basecase, String type) {
+		if(work.isDir())
+		{
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found Dir (" + type +")" + url.toString());
+		    }
+		    //we found a dir
+		    manager.foundDir(url, code, response, basecase, rawResponse, work.getBaseCaseObj());
+		}
+		else
+		{
+		    //found a file
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found File (" + type +")" + url.toString());
+		    }
+		    manager.foundFile(url, code, response, basecase, rawResponse, work.getBaseCaseObj());
+		}
+	}
+	
+
+	private void notifyItemFound(int code, String response, String rawResponse, String basecase) {
+		notifyItemFound(code, response, rawResponse, basecase, "base case");		
+	}
+
+	private void parseHtml(HttpMethodBase httpMethod, String response) {
+		//parse the html of what we have found
+
+	    Header contentType = httpMethod.getResponseHeader("Content-Type");
+
+	    if(contentType != null)
+	    {
+	        if(contentType.getValue().startsWith("text"))
+	        {
+	            manager.addHTMLToParseQueue(new HTMLparseWorkUnit(response, work));
+	        }
+	    }
+		
+	}
+
+	private String getResponseAsString(HttpMethodBase httpMethod) throws IOException {
+		StringBuilder builder = new StringBuilder();
+		
+		try(BufferedReader input = new BufferedReader(new InputStreamReader(httpMethod.getResponseBodyAsStream()))) {
+			builder.append(System.lineSeparator());
+			builder.append(input.readLine());
+		}
+		return builder.toString();
+	}
+
+	private String getHeadersAsString(HttpMethodBase httpMethod) {
+		Header[] headers = httpMethod.getResponseHeaders();
+		
+		StringBuilder builder = new StringBuilder();
+		
+		builder.append(httpMethod.getStatusLine());
+		builder.append(System.lineSeparator());
+		
+		for (Header header : headers) {
+			builder.append(header.getName() + ": " + header.getValue());
```

Comment:
This should be replaced with:
`builder.append(header.getName()).append(": ").append(header.getValue());`
Otherwise it creates an unnecessary `StringBuilder` to do the intermediate concatenations.